microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/snippets/SnippetContextForProperties.java

```diff
@@ -37,31 +41,37 @@
 		],
 		"description": "Configure Quarkus datasource",
 		"context": {
-			"dependency": "quarkus-agroal"
+			"properties": ["quarkus.datasource.url", "quarkus.hibernate-orm.database.generation"]
 		}
 	}
 
  * </code>
  * 
- * This snippet will be available only if "quarkus-agroal" is present as
- * dependency in the project.
+ * This snippet will be available only if
+ * "quarkus.hibernate-orm.database.generation" is present for the project.
  * 
  * @author Angelo ZERR
  *
  */
-public class SnippetContextForProperties implements ISnippetContext<Set<String>> {
+public class SnippetContextForProperties implements ISnippetContext<MicroProfileProjectInfo> {
 
 	public static final TypeAdapter<SnippetContextForProperties> TYPE_ADAPTER = new SnippetContextForPropertiesAdapter();
 
-	private String dependency;
+	private List<String> properties;
 
-	public SnippetContextForProperties(String dependency) {
-		this.dependency = dependency;
+	public SnippetContextForProperties(List<String> properties) {
+		this.properties = properties;
 	}
 
 	@Override
-	public boolean isMatch(Set<String> dependencies) {
-		return dependencies != null && dependencies.contains(dependency);
+	public boolean isMatch(MicroProfileProjectInfo projectInfo) {
+		for (String propertyName : this.properties) {
+			ItemMetadata metadata = MicroProfilePropertiesUtils.getProperty(propertyName, projectInfo);
+			if (metadata != null) {
+				return true;

```

Comment:
According to this documentation from this PR:
>* for `properties files`: the context/properties can be declared in the snippet to show the snippet only if the declared properties (you can use simple String or String array for that) belong to the project.


If `properties` is an array of String, I think we should if each property from the array exists in the project? In this `isMatch` method, it is only checking if at least one property exists. If this was intended, then I think the documentation should be updated to:

```
* for `properties files`: the context/properties (can be String or String array) can be declared by the snippet to show the
snippet only if the declared property belongs to the project. If an array of properties were provided, the snippet will be 
displayed if at least one property belongs to the project.
```

Reply:
Indeed it was not clear. I change my code to match all properties instead of he least one property strategy. I changed README too explain that.